rpc/namespaces/eth/api.go

```diff
@@ -234,6 +240,28 @@ func (api *PublicEthereumAPI) GetBalance(address common.Address, blockNum rpctyp
 		return nil, err
 	}
 
+	if blockNum != rpctypes.PendingBlockNumber {
+		return (*hexutil.Big)(val), nil
+	}
+
+	// update the address balance with the pending transactions value (if applicable)
+	pendingTxs, err := api.backend.PendingTransactions()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, tx := range pendingTxs {
+		if tx == nil {
+			continue
+		}
+
+		if tx.From == address {
+			val = new(big.Int).Sub(val, tx.Value.ToInt())
+		} else if *tx.To == address {
+			val = new(big.Int).Add(val, tx.Value.ToInt())
+		}
```

Comment:
there might be an edge case here where the to and the from are the same, so change this to two `if`s instead of an `if.. else if`